vowpalwabbit/learner.cc

```diff
@@ -207,16 +209,24 @@ class multi_example_handler
 
   void on_example(example* ec)
   {
+    if(ec->malformed)
+    {
+      malformed = true;
+    }
     if (try_complete_multi_ex(ec))
     {
-      _context.template process<multi_ex, learn_multi_ex>(ec_seq);
+      if(!malformed)
+      {
+        _context.template process<multi_ex, learn_multi_ex>(ec_seq);
```

Comment:
malformed examples will never be processed (which is what we want). But a part of the process is calling `finish_example` which, among other things, returns the example back to the pool for reuse. We need to return the discarded examples back to the pool

Reply:
Okay, understood. I will take care of it.